doc/specs/Proto extensions-spec.md

```diff
@@ -38,66 +38,178 @@ in between 2 and 3:
+}
+```
+However, the user is free to make changes to it as usual and those changes will have the 'final say'. 
 
 #### Full profile stubs
 
 Technically, full profile stubs do not need to contain anything (they could just be '\{\}'). However we should
 have some qualifying minimum criteria before we accept a stub as a full profile. I suggest that we only create
 new profile objects from stubs that contain at least the following
 
-* A name (and this name _should not_ match the names of existing profiles as described above, because that would cause us to interpret this profile as modifying an existing one)
+* A name
 * A commandline argument
 * A unique GUID - this is so we avoid conflicts if several different creators name their profile the same thing
 
 As in the case of the dynamic profile generator, if we create a profile that did not exist before (i.e. does not
-exist in the user settings), we need to add the profile to the user settings file and re-save that file. 
+exist in the user settings), we need to add the profile to the user settings file and re-save that file.
+
+Furthermore, we will add a source field to profiles created this way (again, similar to what we do for dynamic profiles).
+The source field value is dependent on how we obtained this json file, and will be touched on in more detail later in the
+spec (see "Creation and location(s) of the json files" - "The source field"). 
+
+Here is an example of a json stub that contains a full profile:
+
+```js
+{
+    "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+    "name": "Cool Profile",
+    "commandline": "powershell.exe",
+    "antialiasingMode": "aliased",
+    "fontWeight": "bold",
+    "scrollbarState": "hidden"
+}
+```
+
+When this profile gets added back to the user's settings file, it will look similar to the way we currently add
+new dynamic profiles to the user's settings file. Going along with the example above, the corresponding
+json stub in the user's settings file will be:
+
+```js
+{
+    "guid": "{a821ae62-9d4a-3e34-b989-0a998ec283e6}",
+    "name": "Cool Profile",
+    "hidden": "false",
+    "source": "local"
+}
+```
+Again, the user will be able to make changes to it as they do for all other profiles. 
+
+#### Creation of colour schemes
+
+As with full profiles, we will have some qualifying criteria for what we accept as full colour schemes: color schemes
+must have
+* A name
+* A background
+* A foreground
+
+This will cause the issue of colour schemes being overridden if there are many creations of a colour scheme with the
+same name. Since for now all we have to uniquely identify colour schemes *is* the name, we will just let this be.
+
+Here is an example of a json stub that contains a colour scheme:
+
+```js
+{
+    "name": "Postmodern Tango Light",
+    "background": '#61D6D6',
+    "foreground": '#E74856'
+}
+```
+
+This stub will *not* show up in the users settings file, similar to the way our default colour schemes do not show up.

```

Comment:
(thinking out loud, not necessarily for the spec) This'll be interesting for the Settings UI, because it's valid to _reference_ this color scheme, but not _modify_ it. The Color Schemes page may have to either...
- remove ColorSchemes from stubs
- present them, but put a note that it's imported and not modifiable
- present them, but when the user modifies one, the modified version gets copied to the settings.json

Reply:
None of the current colour schemes we have show up in the settings file though, how does the settings UI deal with those?

Reply:
Ah. So `GlobalAppSettings` currently has a reference to _all_ of the color schemes. This can't be changed because that's how we look up what a color scheme name actually maps to. The Settings UI reads/edits that instead of the actual JSON directly.

So instead, we need a way to mark where each color scheme came from (like `source` in `Profile`). That way, when the Settings UI presents the Color Schemes page, we handle stub-generated color schemes differently (the list of ideas above).

We _could_ also introduce another top-level navigation item that lets users browse/copy/modify anything generated from stubs. But I'm less of a fan of that idea because it breaks up where all of the color schemes can be found.